File: WEEK 1 - C/pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

//Declaration of the height and the function to check if the number is between 1 and 8.
int check_height(string prompt);
int height = 0;

int main(void)
{
    //Ask the user to introduce a value for the height.
    height = check_height("Height: ");

    //Declaration and assigment of the letter we gonna use for the pyramid.
    char hashtag = '#';

    //Loop for the lines
    for (int rows = 0; rows < height; rows++)
    {
        //Loop for the spaces in each row at the start.
        for (int spaces = height - rows; spaces > 1; spaces--)
        {
            printf(" ");
        }
        //Loop for the left part of the pyramid.
        for (int left_rows = 0; left_rows <= rows; left_rows++)
        {
            printf("%c", hashtag);
        }
        //Spaces in between
        printf("  ");

        //Loop for the right part of the pyramid.
        for (int right_rows = 0; right_rows <= rows; right_rows++)
        {
            printf("%c", hashtag);
        }
        //New Line
        printf("\n");
    }

}

int check_height(string prompt)
{
    do
    {
        height = get_int("%s", prompt);
    }
    while (height < 1 || height > 8);

    return height;
}

File: WEEK 6 - Python/pset6-lab6/sentimental-mario-more/mario.py
height = 0


def main():
    global height
    height = check_height(height)
    draw_pyramids()


def check_height(height):
    while height < 1 or height > 8:
        try:
            height = int(input("Height: "))
        except ValueError:
            print("Oops!  That was no valid number.  Try again...")
    return height


def draw_pyramids():
    # Declaration and assignment of the character we're going to use for the pyramid.
    hashtag = '#'

    # Loop for the lines
    for rows in range(height):
        # Loop for the spaces in each row at the start.
        for spaces in range(height - rows - 1):
            print(" ", end='')

        # Loop for the left part of the pyramid.
        for left_rows in range(rows + 1):
            print(hashtag, end='')

        # Spaces in between
        print("  ", end='')

        # Loop for the right part of the pyramid.
        for right_rows in range(rows + 1):
            print(hashtag, end='')

        # New line
        print()


main()
